azkaban-common/src/main/java/azkaban/imagemgmt/permission/PermissionManagerImpl.java

```diff
@@ -0,0 +1,148 @@
+   */
+  private static final Permission MEMBER_PERMISSION = new Permission(Type.CREATE, Type.GET,
+      Type.UPDATE, Type.DELETE);
+  /**
+   * The default permission is only GET access to the image management APIs.
+   */
+  private static final Permission READ_ONLY_PERMISSION = new Permission(Type.GET);
+
+  @Inject
+  public PermissionManagerImpl(final ImageTypeDao imageTypeDao) {
+    this.imageTypeDao = imageTypeDao;
+  }
+
+  /**
+   * Checks the permission based on image type name, user id and Permission type.
+   *
+   * @param imageTypeName
+   * @param userId
+   * @param type
+   * @return boolean
+   */
+  @Override
+  public boolean hasPermission(final String imageTypeName, final String userId, final Type type) {
+    // Gets the image type metadata including ownerships.
+    final Optional<ImageType> optionalImageType =
+        this.imageTypeDao.getImageTypeWithOwnershipsByName(imageTypeName);
+    boolean hasPermission = false;
+    if (optionalImageType.isPresent()) {
+      final ImageType imageType = optionalImageType.get();
+      if (!CollectionUtils.isEmpty(imageType.getOwnerships())) {
+        // Gets the permission of the user based on image type ownership metadata.
+        final Optional<Permission> optionalPermission = getPermission(userId,
+            imageType.getOwnerships());
+        // Check if the given Permission.Type contains in the permission of the user.
+        // Check if the the user permission contains Permission.Type ADMIN.
+        if (optionalPermission.isPresent() && (optionalPermission.get().isPermissionSet(type)
+            || optionalPermission.get().isPermissionSet(Permission.Type.ADMIN))) {
+          hasPermission = true;
+        }
+      } else {
+        log.error(String.format("API access permission check failed. There is no ownership record "
+            + "for image type: %s.", imageTypeName));
+        throw new ImageMgmtException(String.format("API access permission check failed. There is "
+            + "no ownership record for image type: "
+            + "%s.", imageTypeName));
+      }
+    } else {
+      log.error(
+          String.format("API access permission check failed. The image type metadata not found "
+              + "for image type: %s.", imageTypeName));
+      throw new ImageMgmtException(String.format("API access permission check failed. The image "
+              + "type metadata not found for image type: %s.",
+          imageTypeName));
+    }
+    return hasPermission;
+  }
+
+  /**
+   * Gets the permission for the given user. Checks the permission of the given user based on image
+   * ownership metadata.
+   *
+   * @param userId
+   * @param imageOwnershipList
+   * @return Optional<Permission>
+   */
+  private Optional<Permission> getPermission(final String userId,
+      final List<ImageOwnership> imageOwnershipList) {
+    for (final ImageOwnership imageOwnership : imageOwnershipList) {
+      if (userId.equals(imageOwnership.getName())) {
```

Comment:
Nit: imageOwnership.getName() is inappropriate when you are comparing userId. The name of method should be getUserId in imageOwnership.

Reply:
There was a mistake. Corrected to owner.